### PR TITLE
GrafanaRoute: Use React.Lazy instead of react-loadable and improve error handling

### PR DESCRIPTION
**Pull request by torkelo**
fixes 
Screenshot from 2022-09-17 16-31-40

* Removes use of react-loadable and replaces it with Suspense + React.lazy 
* Moves chunk error handling to be handled via ErrorBoundary (is the recommended way)
* Moves page error handling and loading fallback to GrafanaRoute
* Adds new tests for GrafanaRoute 

**Fixes a pretty major issue**: 
Wit this PR you still have NavBar or TopNav when a page crashes (due to ErrorBoundary on the Page level, before this PR we only had an ErrorBoundary at the top app level above the nav components). So in main when a page errors you have no way to navigate to another page. But with this PR you have nav on page error :tada: 

**Follow ups**
* Want to improve the loading state in later PRs, maybe explore a loading state in top nav bar (in addition to page spinner) to indicate a loading state 
* The error page currently does not know the navId, could be worth exploring moving navId to route again and solve the issues that had as then we can use it on the error page (and show the correct section and breadcrumb)

### CODE REVIEW

File: public/app/core/components/DynamicImports/SafeDynamicImport.tsx [hunk 1]
```diff
@@ -1,27 +1,5 @@
...
-
-  return null;
-};
-
-export const SafeDynamicImport = (loader: () => Promise<any>): GrafanaRouteComponent =>
-  Loadable({
-    loader: loader,
-    loading: loadComponentHandler,
-  });
+export const SafeDynamicImport = (loader: () => Promise<any>): GrafanaRouteComponent => React.lazy(loader);
```
Comment (torkelo) [inline public/app/core/components/DynamicImports/SafeDynamicImport.tsx:5]: I tried to replace all uses of SafeDynamicImport with React.lazy, it works at runtime, but typescript is complaining for some uses. Typescript is having some issues converting the components our pages return to `GrafanaRouteComponent`, when we have connector(withTheme2(Component))  and other issues when just using the page component has typed queryParams. Think we can address those in a different PR if we want to improve typings.

**Review by kaydelaney — APPROVED** [review]
Nice!

**Review by ashharrison90 — APPROVED** [review]
this looks great!

1 minor nit:
- think we can remove `react-loadable`/`@types/react-loadable` from our `package.json` now

File: public/app/core/navigation/GrafanaRouteLoading.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+const getStyles = () => ({
+  loadingPage: css({
+    height: '100%',
+    flexDrection: 'column',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  }),
```
Comment (dprokop) [inline public/app/core/navigation/GrafanaRouteLoading.tsx:23]: Nit/question: Any benefit of using `css({...})` rather than templated srting?

**Review by dprokop — APPROVED** [review]
Tested this and haven't found any issues with navigation 👍

File: public/app/core/navigation/GrafanaRouteLoading.tsx (hunk 2 above)
Comment (torkelo) [reply public/app/core/navigation/GrafanaRouteLoading.tsx:23]: @dprokop speed (no need to parse it), better autocomplete / IntelliSense, typechecking , easier to write and compose, I use it for all new styles

**Pull request merged**